pypika/queries.py

```diff
@@ -423,12 +432,27 @@ def from_(self, selectable):
 
             When a ``str`` is passed, a table with the name matching the ``str`` value is used.
 
+        :param correlated_table:
+            Type: ``Table``, ``Query``, or ``str``
+
+            An instance of the outer table (or string table name)
+            used if this query is forming part of a correlated subquery.
+            (a subquery inside SELECT or WHERE).
+
         :returns
             A copy of the query with the table added.
         """
 
         self._from.append(Table(selectable) if isinstance(selectable, str) else selectable)
 
+        if correlated_table is not None:
+            # add a reference to the outer table in a correlated subquery.
+            self._correlated.append(
+                Table(correlated_table)
```

Comment:
Need to consider the case that the correlated value is another Query, not a Table. Should probably add a test for that case. Instances of Tables and Queries always need to be interchangeable.